components/event-publisher-proxy/testing/utils.go

```diff
@@ -163,11 +164,33 @@ func WaitForHandlerToStart(t *testing.T, healthEndpoint string) {
 
 // GeneratePortOrDie generates a random 5 digit port or fail
 func GeneratePortOrDie() int {
-	port, err := generatePort()
-	if err != nil {
-		log.Fatalf("Failed to generate port with error: %v", err)
+	tick := time.NewTicker(time.Second / 2)
+	defer tick.Stop()
+
+	timeout := time.NewTimer(time.Minute)
+	defer timeout.Stop()
+
+	for {
+		select {
+		case <-tick.C:
+			{
+				port, err := generatePort()
+				if err != nil {
+					break
+				}
+
+				if err := check(port); err != nil {
```

Comment:
When ignoring the returned error a simple kind of `isPortUnused(port) bool` would be enough an more speaking (otherwise question would have to be what is checked).

Reply:
Done.